src/components/TaskCard.js

```diff
@@ -0,0 +1,57 @@
+import { useState } from "react";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import Typography from "@mui/material/Typography";
+import CardActions from "@mui/material/CardActions";
+import IconButton from "@mui/material/IconButton";
+import CheckIcon from "@material-ui/icons/Check";
+import DeleteIcon from "@material-ui/icons/Delete";
+import EditIcon from "@material-ui/icons/Edit";
+import { green } from "@mui/material/colors";
+
+function TaskCard({ task, onDelete, onEdit, onCompleteChange }) {
+  const [completed, setCompleted] = useState(false);
+
+  function formatDate(date) {
+    const options = { year: "numeric", month: "numeric", day: "numeric" };
+    return new Date(date).toLocaleDateString("en-US", options);
+  }
+
+  const handleComplete = () => {
+    setCompleted(!completed);
```

Comment:
always user setter function when updating based on previous value.  
Read and understand why here:  
https://react.dev/reference/react/useState#setstate

Reply:
Anyway, this state is redundant.